cannon/example/multithreaded/go.mod

```
@@ -0,0 +1,3 @@
+module multithreaded
+
+go 1.20
```

nit: might as well use the latest used in the monorepo
```suggestion
go 1.21
```